runtime/oti/j9nonbuilder.h

```diff
@@ -1472,7 +1472,13 @@ typedef struct J9ExceptionHandler {
 	U_32 exceptionClassIndex;
 } J9ExceptionHandler;
 
-#if defined(__xlC__) || defined(J9ZOS390)  /* Covers: Z/OS, AIX, Linux PPC*/
+/*
+ * For Linux PPC LE:
+ * __ibmxlc__ macro is defined for xlc 13.1.1 and xlc 16.1.1
+ * __xlC__ macro is defined only when -qxlcompatmacros compiler option is enabled (for xlc 16.1.1 only)
+ */
```

Comment:
I'd like to see this comment expanded: it's helpful to know when those macros are defined, but it doesn't explain why we need the proposed change or what would go wrong without it.

Reply:
I expanded the comment.

Reply:
The comment suggests that the behavior of xlc 16.1.1 differs from earlier versions: why is there no version check here?

Reply:
Yes, the behavior changed in xlc 16.1.1.
`#pragma pack(1)` followed by `#pragma pack(pop)` generates an warning:
```
�[1m../../../oti/j9nonbuilder.h:1498:9: �[0m�[0;1;31merror: �[0m�[1m#pragma pack(pop, ...) failed: stack
      empty [-Werror,-Wignored-pragmas]�[0m
#pragma pack(pop) 
```

As per https://www.ibm.com/support/knowledgecenter/SSXVZZ_16.1.1/com.ibm.xlcpp1611.lelinux.doc/migrate/migration_earlier_version.html

> Starting from IBM XL C/C++ for Linux, V13.1.6, `-qnoxlcompatmacros` is set to default. To define these legacy macros: `__xlC__`, `__xlC_ver__`, `__IBMCPP__` (C++ only), `__IBMC__`, and `__xlc__` (C only) , you must explicitly specify `-qxlcompatmacros`.

Note that`-qxlcompatmacros` is enabled for Linux PPC LE (and AIX) 16.1.1. Below are the compiler macros per platform: 


  | xlc 12 | xlc 13 | xlc16
-- | -- | -- | --
AIX | C/C++ | _C/C++ | xlclang/xlclang++
  | `__xlC__` defined | `__xlC__` defined | `__xlC__` defined
  | `__ibmxl__` undefined | __ibmx__undefined | `__ibmxl__` defined
Linux PPC LE |   | _C/C++ | _C/C++
  |   | `__xlC__` defined | `__xlC__` defined
  |   | `__ibmxl__` undefined | `__ibmxl__` defined
Linux PPC BE |   | _C/C++ |  
  |   | `__xlC__` defined |  
  |   | `__ibmxl__` undefined |  




   